File: backend/generate_sample.js
const fs = require('fs');
const path = require('path');

function generateSampleData() {
    const startDate = new Date('2022-01-01T00:00:00Z');
    let csvContent = 'date,value\n';
    
    // 730 days
    for (let i = 0; i < 730; i++) {
        const currentDate = new Date(startDate.getTime() + (i * 24 * 60 * 60 * 1000));
        const dateStr = currentDate.toISOString().split('T')[0];
        
        // Base trend
        const base = 100 + (i * (200 / 730)); // from 100 to 300
        
        // Weekly Seasonality (0 = Sunday, 6 = Saturday)
        const day = currentDate.getDay();
        const seasonal = (day === 0 || day === 6) ? 50 : 10;
        
        // Noise
        const noise = (Math.random() - 0.5) * 30; // -15 to +15
        
        let val = base + seasonal + noise;
        
        // Random spikes 1% chance
        if (Math.random() < 0.01) {
            val += (Math.random() * 100) + 50;
        }
        
        val = Math.max(0, val).toFixed(2);
        csvContent += `${dateStr},${val}\n`;
    }
    
    const dir = path.join(__dirname, '..', 'sample_data');
    if (!fs.existsSync(dir)){
        fs.mkdirSync(dir, { recursive: true });
    }
    
    fs.writeFileSync(path.join(dir, 'sales_data.csv'), csvContent);
    console.log("Sample data generated successfully using Node!");
}

generateSampleData();

File: backend/generate_sample.py
import numpy as np
import pandas as pd
from datetime import datetime, timedelta
import os

def generate_sample_data():
    start_date = datetime(2022, 1, 1)
    dates = [start_date + timedelta(days=i) for i in range(730)]  # 2 years
    
    # Base trend (upward)
    base = np.linspace(100, 300, 730)
    
    # Weekly seasonality
    # e.g. higher on weekends
    seasonal = []
    for d in dates:
        if d.weekday() >= 5: # Sat, Sun
            seasonal.append(50)
        else:
            seasonal.append(10)
            
    # Add some random noise
    noise = np.random.normal(0, 15, 730)
    
    # Create a few spikes
    spikes = np.zeros(730)
    spike_indices = np.random.choice(730, size=10, replace=False)
    for idx in spike_indices:
        spikes[idx] = np.random.uniform(50, 150)
        
    values = base + seasonal + noise + spikes
    values = np.maximum(values, 0) # Ensure no negative sales
    
    df = pd.DataFrame({
        'date': dates,
        'value': values.round(2)
    })
    
    os.makedirs('../sample_data', exist_ok=True)
    df.to_csv('../sample_data/sales_data.csv', index=False)
    print("Sample data generated!")

if __name__ == "__main__":
    generate_sample_data()
